Sources/MapboxMaps/Foundation/MapboxObservable.swift

```diff
@@ -92,6 +96,27 @@ internal final class MapboxObservable: MapboxObservableProtocol {
         return cancelable
     }
 
+    internal func onNext<Payload>(event: MapEvents.Event<Payload>, handler: @escaping (MapEvent<Payload>) -> Void) -> Cancelable {
+        let cancelable = CompositeCancelable()
+        let observer = BlockObserver {
+            handler(MapEvent(event: $0))
+            cancelable.cancel()
+        }
+        subscribe(observer, events: [event.name])
+        // Capturing self and observer with weak refs in the closure passed to BlockCancelable
+        // avoids a retain cycle. MapboxObservable holds a strong reference to observer, which has a
+        // strong reference to cancelable, which has a strong reference to BlockCancelable, which only
+        // has weak references back to MapboxObservable and observer. If MapboxObservable is deinited,
+        // observer will be released.
+        cancelable.add(BlockCancelable { [weak self, weak observer] in
+            if let self = self, let observer = observer {
+                self.unsubscribe(observer, events: [])
+            }
+        })
+        return cancelable
```

Comment:
As we are going to support deprecated methods for a while, can we refactor it to reduce duplicated code?
Third function with `[String]` as an argument looks nice.

Reply:
Nice idea! Refactored.